packages/ses-ava/src/ses-ava-test.js

```diff
@@ -100,27 +163,27 @@ const wrapTester = (testerFunc, logger = defaultLogger) => {
  * Then the calls to `test` in the rest of the test file will act like they
  * used to, except that, if a test fails because the test function (the
  * callback argument to `test`) throws or returns a promise
- * that eventually rejects, the error is first sent to the `console` before
- * propagating into `rawTest`.
+ * that eventually rejects, the error is first sent to the SES-aware `console`
+ * before propagating into `rawTest`.
  *
- * @param {TesterInterface} avaTest
- * @param {Logger} [logger]
- * @returns {TesterInterface}
+ * @template {TesterFunc} T Ava `test`
+ * @param {T} avaTest
+ * @param {Logger} [logError]
+ * @returns {T}
  */
-const wrapTest = (avaTest, logger = defaultLogger) => {
-  const testerWrapper = /** @type {TesterInterface} */ (
-    wrapTester(avaTest, logger)
-  );
-  for (const methodName of testerMethodsWhitelist) {
-    if (methodName in avaTest) {
-      /** @type {TesterFunc} */
-      const testerMethod = (title, implFunc, ...otherArgs) =>
-        avaTest[methodName](title, implFunc, ...otherArgs);
-      testerWrapper[methodName] = wrapTester(testerMethod);
-    }
+const wrapTest = (avaTest, logError = defaultLogError) => {
+  const sesAvaTest = augmentLogging(avaTest, logError);
+  for (const methodName of overrideList) {
+    sesAvaTest[methodName] = augmentLogging(avaTest[methodName], logError);
   }
-  harden(testerWrapper);
-  return testerWrapper;
+  harden(sesAvaTest);
+  return sesAvaTest;
 };
-// harden(wrapTest);
+// Most instantiations of this module are pre-lockdown,
+// in which case harden is not ready here.
+try {
+  harden(wrapTest);
+} catch (err) {
+  freeze(wrapTest);
```

Comment:
If `freeze` is sufficient, why bother with `harden`?

If we need `harden` before lockdown, couldn't this just have its own?
```
const harden = makeHardener();
```

Reply:
I don't think so, because that would traverse and mutate intrinsics in precisely the way this is trying to avoid.

Reply:
Given that a call to `wrapTest` has `harden`, I don't understand what is gained by `harden` here vs `freeze`. Not a blocker but please consider simplifying this to `freeze` (if appropriate) or leaving a comment on what will be different if only `freeze` is run.

Reply:
Any thoughts, @erights? This stems from https://github.com/Agoric/agoric-sdk/pull/5774#discussion_r922724162 .

Reply:
I've tentatively opted for unconditional `freeze` with preceding justification.

Reply:
@michaelfig @kriskowal , are you two were exploring a pre-lockdown `harden` that just registers things to be hardened once `lockdown` is called? I seem to recall that on some thread somewhere. What's the status of that?

In the absence of a pre-lockdown registering `harden`, I have taken to manually freezing everything that a post-lockdown `harden` would have frozen. Another good thing about arrow functions is that is only the function itself. (Also true for concise methods, but not for `function` functions nor classes.)

So yes, if `wrapTest` is an arrow function, then just a `freeze` is enough.

> If `freeze` is sufficient, why bother with `harden`?

- fragility. Were `wrapTest` changed from an arrow function to a `function` function. Or were more properties added to it before freezing/hardening, `freeze` would fail to freeze more things than `harden` would fail to freeze.
- memoizing. `harden` remembers what's hardened so it can avoid redundant work on future `harden`s.

Reply:
@erights keeping in mind that this is code to be imported by test files, is that an argument for preferring `try { harden(wrapTest) }` over an unconditional `freeze(wrapTest)`?

Reply:
I don't see why. Why would use by test files suggest this?

Reply:
I think Richard's asking if you object to the use of `freeze` instead of `harden` for the `wrapTest` function. I.e. are you requesting changes to this PR.

I asked a sort of academic question about the difference between `freeze` and `harden` and you contrasted them, but it wasn't clear whether you assert that those differences imply the need for the (conditional) use of `harden` here.

I'll make the case that they don't. Memoization is irrelevant because this happens once at module load. "Fragility" is only a risk that a change could be made to wrapTest that would freeze fewer things. As Richard points out, this is a testing util. Test environments are always more malleable than production. `wrapTest` is meant to test code that runs in SES, but we don't need to defend against mutations to wrapTest because it's our tool. If we don't trust our own tests then we've got bigger problems :)  Is this about a coding discipline or to defend against some real risk? Perhaps I'm just not being creative enough… is the attack that some package gets into our import graph and itself imports `ses-ava` and mutates wrapTest to deceive our other tests into satisfying some important security property when they don't?

Reply:
@turadg 100%. Sorry I was unclear. I was indeed saying three things:
    * approving the use of `freeze` rather than `harden` specifically here in this PR
    * that I see no need for a condition harden or a harden in a try/catch here in this PR
    * why, in general, we should avoid freeze and use harden instead. Where that observation was not intended to apply to this case.